hardware/is_camera/src/camera.cc

```diff
@@ -288,18 +294,43 @@ namespace is_camera {
       exit(EXIT_FAILURE);
     }
 
-    if (bayer_enable_) {
-      if (!camera.GetUInt("bayer_throttle_ratio", &bayer_throttle_ratio_)) {
-        FF_FATAL("Bayer throttle ratio not specified.");
-        exit(EXIT_FAILURE);
-      }
+    if (!camera.GetUInt("bayer_throttle_ratio", &bayer_throttle_ratio_)) {
+      FF_FATAL("Bayer throttle ratio not specified.");
+      exit(EXIT_FAILURE);
     }
 
     if (thread_running_) {
       v4l_->SetParameters(camera_gain_, camera_exposure_);
     }
   }
 
+  // Enable or disable the feature timer
+  bool CameraNodelet::EnableBayerService(ff_msgs::SetBool::Request & req,
+                     ff_msgs::SetBool::Response & res) {
+    if (req.enable && !bayer_enable_) {
+      bayer_camera_topic_ = camera_topic_ + TOPIC_HARDWARE_CAM_SUFFIX_BAYER_RAW;
```

Comment:
Right now, bayer variable setup/teardown happens in multiple places, with a lot of code copy/pasted. It makes it harder to understand all the states the variables could be in.

I would recommend creating a single function that can be called both at node initialization and at service enable/disable so that all the setup/teardown goes through the one code path.